02_bubble/nicholeL/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+//Compare two adjacent numbers
+func bubbleSort(s []int) []int {
+	for index := 0; index < len(s); index++ {
+		for count := 0; count < len(s)-index-1; count++ {
+			if s[count] >= s[count+1] {
+				s[count], s[count+1] = s[count+1], s[count]
+			}
+		}
+	}
+	return s
+}
+
+func insertSort(s []int) []int {
+
+	for first := 1; first < len(s); first++ {
+		index := first
+		temp := s[first]
```

Comment:
keep assignments of related things together where possible.
from our reference solution, which makes the shift right of the elements from "insertion point" to what you call `first` more obvious:
```go
	for i := 1; i < len(s); i++ {
		var j int
		for j = i; j > 0; j-- {
			if s[j-1] <= s[i] {
				break // We've found the insertion point
			}
		}
		// insert s[i] at s[j] moving the rest up
		tmp := s[i]
		copy(s[j+1:i+1], s[j:i])
		s[j] = tmp
	}
```

Reply:
thanks for your advice and will update